dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/oracle/AbstractOracleConnector.java

```diff
@@ -106,7 +110,8 @@ public Handle open(ConnectorArguments arguments) throws Exception {
         url = "jdbc:oracle:thin:@//" + host + ":" + port + "/" + arguments.getOracleServicename();
       }
     }
-    // LOG.info("URL IS " + url);
+    url = String.format("%s?%s", url, USE_FETCH_SIZE_WITH_LONG_COLUMN_ARG);
```

Comment:
1. It seems that the question mark is available only in `Easy Connect Plus` Oracle JDBC URLs that work only since Oracle 19c.
https://docs.oracle.com/en/database/oracle/oracle-database/21/jjdbc/data-sources-and-URLs.html#GUID-C7D48803-09E7-42C5-A0F6-32B009D0C20C

    The older versions used `new OracleDataSource().setConnectionProperties(prop)`. Would it make sense to support all the options - 1. existing code for backwards-compatibility, 2. the Oracle 19c compatible code and 3. the `setConnectionProperties`?

2. If the URL is provided in the command-line, it can already contain the `?` character. The concatenation of a new property should take it into account and use an `&` character in such a case. Also, the property can already be included in the URL - setting the flag to either false or true. The property should not be included twice. It also probably should not change the value provided in the command-line.
https://docs.oracle.com/en/database/oracle/oracle-database/21/jjdbc/data-sources-and-URLs.html#GUID-843F6EC2-EDFD-4A88-98DE-B45E1CEADC0E

Reply:
I changed the approach to set the connection properties instead of modifying URL. I tested it with the URL that already includes this property and it has a higher priority.